### ISSUE TITLE
Learning Disentangled Joint Continuous and Discrete Representations

### ISSUE THREAD

**Issue by koukyo1994** [koukyo1994/awesome-disentangled-representations#1]
# Learning Disentangled Joint Continuous and Discrete Representations

Emilien Dupont
NeurIPS2018

## What is this about?

A proposal of Unsupervised Disentangled Representation(UDR) method that can learn both continuous and discrete representations. To do this, they use a relaxed discrete distribution for prior and also control the amount of information encoded in each latent variable.

## What's better than previous works?

While β-VAE (and other works of UDR) can only learn continuous variables, this work can learn both discrete and continuous variables.

## What's the key technique?

In the proposed method, they split the latent variables in to two parts: continuous variables z and discrete variables c and use loss function (similar to β-VAE with capacity)

![](https://latex.codecogs.com/png.latex?%5Cinline%20%5Cmathcal%7BL%7D%28%5Ctheta%2C%5Cphi%29%3D%5Cmathbb%7BE%7D_%7Bq_%5Cphi%28%5Cmathbf%7Bz%2Cc%7Cx%7D%29%7D%5B%5Clog%20p_%5Ctheta%28%5Cmathbf%7Bx%7Cz%2Cc%7D%29%5D-%5Cgamma%20%7CD_%7BKL%7D%28q_%5Cphi%28%5Cmathbf%7Bz%7Cx%7D%29%5C%7Cp%28%5Cmathbf%7Bz%7D%29%29-C_z%7C-%5Cgamma%7CD_%7BKL%7D%28q_%5Cphi%28%5Cmathbf%7Bc%7Cx%7D%29%5C%7Cp%28%5Cmathbf%7Bc%7D%29%29-C_c%7C)

For discrete variables, they use a product of independent Gumbel Softmax distributions to parameterize q(c|x) and used Gumbel Max trick to sample from it. In the equation below, y is continuous approximation of c:

![](https://latex.codecogs.com/png.latex?%5Cdpi%7B100%7D%20y_k%3D%5Cfrac%7B%5Cexp%28%28%5Clog%5Calpha_k&plus;g_k%29/%5Ctau%29%7D%7B%5Csum_i%5Cexp%28%28%5Clog%5Calpha_i&plus;g_i%29/%5Ctau%29%7D%7E%7Eg_k%5Csim%5Cmathrm%7BGumbel%7D%280%2C1%29%7E%7E%5Ctau%3A%5Cmathrm%7Btemperature%7D)

## How did they evaluate that?

Qualititative evaluation with MNIST, FashionMNIST, CelebA, Chairs using latent traversal + Quantitative evaluation with dSprites using β-VAE score.

## Any discussions & insigts?

The proposed method outperforms previous works when a discrete generative factor is prominent.

## Which paper to read next?

Esmaeili, Babak, et al. "Structured disentangled representations." The 22nd International Conference on Artificial Intelligence and Statistics. PMLR, 2019.

![arch](https://www.researchgate.net/profile/Emilien_Dupont/publication/324167090/figure/fig7/AS:663199896567813@1535130585312/JointVAE-architecture-The-input-x-is-encoded-by-q-into-the-parameters-of-the-latent.png)
